modules/dnn/src/webgpu/include/tensor.hpp

```diff
@@ -0,0 +1,52 @@
+#ifndef OPENCV_DNN_WEBGPU_TENSOR_HPP
+#define OPENCV_DNN_WEBGPU_TENSOR_HPP
+// #ifdef HAVE_WEBGPU
+#include<dawn/webgpu_cpp.h>
+// #endif
+#include"buffer.hpp"
+#include"wgpucom.hpp"
+#include<memory>
+namespace cv { namespace dnn { namespace webgpu {
+// #ifdef HAVE_WEBGPU
+class Buffer;
+class Tensor{
+public:
+    Tensor(Format fmt = wFormatFp32);
+    Tensor( const void* data, size_t size_in_byte, 
+            wgpu::BufferUsage usage, 
+            Format fmt = wFormatInt32);  //uniform buffer
+    Tensor( const void* data, std::vector<int>& shape, 
+            wgpu::BufferUsage usage, 
+            Format fmt = wFormatFp32);
+    const void* map();
```

Comment:
Since this map() can only support "read", use function name mapRead() to make this explicitly.